### DISCUSSION TITLE
Add Toast.IsShown property

### DISCUSSION THREAD

**Discussion by CodingOctocat** [CommunityToolkit/Maui#1909] [New Feature Discussions]
I'm trying to implement a "press again to exit" feature when I'm on the homepage. If the user presses the back button while on the home page, then a Toast pops up prompting to press it again, and if the user presses the back button again while the Toast is being displayed, then the App exits, and I need something like a `SnackBar.IsShown` property to monitor the display state of the Toast.

My current alternatives:
1. Use SnackBar
2. Still use Toast and record the time, `ToastDuration.Short` defaults to 2s (I'm not sure if the API will change in the future)

**Comment by VladislavAntonyuk** [comment]
It is more reliable to store the LastBackButtonClickedTime. In that case you don't need to depend on 3rd party property value that may be changed.

**Reply by CodingOctocat** [reply]
I am using the following code and it does work well and `_toastDurationShort = 2s`, matches the display time of Toast(Short). The only downside is that I need to update the code if `_ToastDuration.Short` changes in the future.

AppShell.xaml.cs

```csharp
protected override async void OnNavigating(ShellNavigatingEventArgs args)
{
    base.OnNavigating(args);

    // Cancel any back navigation
    if (args.Source == ShellNavigationSource.Pop)
    {
        if (DateTime.Now - _toastShownTime < _toastDurationShort)
        {
            await _quitToast.Dismiss();

            return;
        }

        args.Cancel();

        await _quitToast.Show();
        _toastShownTime = DateTime.Now;
    }
```

**Reply by VladislavAntonyuk** [reply]
as the only Android has the back button, I suggest you override the behavior in MainActivity and use native android toast, not from the MCT.

**Reply by CodingOctocat** [reply]
Thanks for the advice.

**Reply by CodingOctocat** [reply]
I overlooked an article, This article solved my problem: [Migrate the deprecated OnBackPressed function in .NET MAUI Android application](https://vladislavantonyuk.github.io/articles/Migrate-the-deprecated-OnBackPressed-function-in-.NET-MAUI-Android-application/)

**Reply by VladislavAntonyuk** [reply]
This is mine. Happy to help!
